core/execution/market.go

```diff
@@ -2612,7 +2557,10 @@ func (m *Market) amendOrder(
 			if needToRemoveExpiry {
 				m.expiringOrders.RemoveOrder(expiresAt, existingOrder.ID)
 			}
-			if needToAddExpiry {
+			// need to make sure the order haven't been matched with the
+			// amend, consuming the remain volume as well or we would
+			// add an order while it's not needed to the expiring list
+			if needToAddExpiry && cnf != nil && !cnf.Order.IsFinished() {
```

Comment:
I suppose we would have added order in expiry list which might have traded fully due to the amend...